processor/src/main/kotlin/permissions/dispatcher/processor/util/Extensions.kt

```diff
@@ -122,15 +122,19 @@ fun FileSpec.Builder.addTypes(types: List<TypeSpec>): FileSpec.Builder {
 fun TypeName.correctJavaTypeToKotlinType(): TypeName {
     return if (this is ParameterizedTypeName) {
         val typeArguments = this.typeArguments.map { it.correctJavaTypeToKotlinType() }.toTypedArray()
-        return this.rawType.parameterizedBy(*typeArguments)
+        val rawType = ClassName.bestGuess(this.rawType.correctJavaTypeToKotlinType().toString())
```

Comment:
why do we need this line change? 👀

Reply:
Because collection is a parametrized type.
Without that change transition will look like:
`java.util.Set<java.lang.String>` -> `java.util.Set<kotlin.String>`.

It's not enough just to convert `typeArguments`, we have to convert collection itself (`rawType`).